mozilla/activity-stream
# Bug 1560065 - Refactor first run for skyline, replace modal onboarding with triplets

### Scope
In this PR I refactored the first run experience to:
* Remove the old modal onboarding code
* Make the triplets independent of the trailhead modal / i.e. work with return to AMO and the overlay experience (so we can experiment with any combination)
* Move some of the shared logic around hiding/showing/adding fluent files etc. into a single `FirstRun` component.

I wrote it with hooks initially (see ff9a667) but I encountered some weird test failures around portals that I think are because of enzyme. Maybe in a future PR we could try restoring it.

### Basic structure

Whereas the old code in `asrouter-content.jsx` had different methods for rendering the various onboarding experiences, I've refactored it to have a more component-based structure

```jsx
<FirstRun> 
  <Interrupt />
  <Triplets />
</FirstRun>
```
Where `Interrupt` contains the logic for whether to show the fxa overlay, return to amo, or trailhead,  and `Triplets` renders the cards on the new tab page. Both can be omitted.

### Manual testing

You can test all the variants by changing the `trailhead.firstrun.branches` pref and restarting the browser. Please test the following values:

* `join-privacy`: the trailhead modal + triplets
* `sync`: just trailhead modal
* `control`: This is just the fxa overlay
* `control-privacy`: fxa overlay + triplets

To test return to amo, open `about:newtab#devtools-targeting`, scroll to the bottom and press Force Attribution, and restart the browser. If you have `join-privacy` or `control-privacy` set, you should see triplets.

cc @AllegroFox @JeaneC if you'd like to help review!

### TODO
* [ ] Remove strings for old onboarding stuff
